de.metas.serviceprovider/src/main/java/de/metas/serviceprovider/issue/S_Issue.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * #%L
+ * de.metas.serviceprovider.base
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.serviceprovider.issue;
+
+import de.metas.serviceprovider.external.reference.ExternalReferenceRepository;
+import de.metas.serviceprovider.external.reference.ExternalReferenceType;
+import de.metas.serviceprovider.model.I_S_Issue;
+import lombok.NonNull;
+import org.adempiere.ad.modelvalidator.annotations.Interceptor;
+import org.adempiere.ad.modelvalidator.annotations.ModelChange;
+import org.compiere.model.ModelValidator;
+import org.springframework.stereotype.Component;
+
+@Interceptor(I_S_Issue.class)
+@Component
+public class S_Issue
+{
+	private final ExternalReferenceRepository externalReferenceRepository;
+
+	public S_Issue(final ExternalReferenceRepository externalReferenceRepository)
+	{
+		this.externalReferenceRepository = externalReferenceRepository;
+	}
+
+	@ModelChange(timings = ModelValidator.TYPE_AFTER_DELETE)
+	public void afterDelete(@NonNull final I_S_Issue record)
+	{
+		externalReferenceRepository.deleteByRecordIdAndType(record.getS_Issue_ID(), ExternalReferenceType.ISSUE_ID);
```

Comment:
not 100% sure..did you check if `record.getS_Issue_ID()` is still set to be `>0` at this point? might be just out of habit, but i think i would do this on `TYPE_BEFORE_DELETE`. Anyways if the delete fails elsewehere, the transaction will be rolled back, so the external reference won't be gone

Reply:
yep, the ID is still there, but now there I think about it, it's better to have the removal before delete as we could add some triggers on the referenced tables in order to prevent inconsistent
delete operations done directly in DB.